# Profitable App Profiles for the App Store and Google Play Markets

For this project, we'll pretend we're working as data analysts for a company that builds Android and iOS mobile apps. We make our apps available on Google Play and the App Store.

What will be done in this project is to analyze whether a free application can generate profits or not? The applications analyzed are sourced from the App Store and Google Play. What do we do in this project is a free whether analysis app in Google Play and App Store is profitable.

At the end of the project, we want to make developers understand what types of apps are likely to attract more users.

The original solution is [here](https://github.com/dataquestio/solutions/blob/master/Mission350Solutions.ipynb).

# Data

## Acquisition

Collect and analyze data about mobile apps available on Google Play and the App Store

[A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play collected in August 2018. You can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

[A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7,000 iOS apps from the App Store collected in July 2017. You can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

# Importing

First step, reader function is imported from csv modele, then assign it to variable.

In [1]:
from csv import reader

# Open AppleStore.csv as appstore
# To prevent UnicodeDecodeError, add encoding="utf8" to the open() function
#
opened_file = open('AppleStore.csv', encoding='utf8')
read_file = reader(opened_file)
appstore = list(read_file)
appstore_header = appstore[0]
appstore = appstore[1:]

# Open googleplaystore.csv as gplay
#
opened_file = open('googleplaystore.csv', encoding='utf8')
read_file = reader(opened_file)
gplay = list(read_file)
gplay_header = gplay[0]
gplay = gplay[1:]

## Exploring

In [2]:
# We'll start by opening and exploring these two data sets. 
# To make them easier for you exploring it,
# we created a function named explore_data()
#
def explore_data(dataset, start, end, rows_and_columns=False):
    ''' 
    dataset, expected to be a list of lists.
    start and end, expected to be integers and represent the starting and the ending indices of a slice from the data set.
    rows_and_columns, expected to be a Boolean and has False as a default argument.
    '''
    
    dataset_slice = dataset[start:end] # slice the dataset    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line between rows
        
    if rows_and_columns:
        print('Number of rows:', len(dataset)) # will return row(s) count
        print('Number of columns:', len(dataset[0])) # will return column(s) count

### Exploring Apple Store Dataset

Using explore_data() above, we'll print the column names and first three rows of Apple Store dataset.

In [3]:
print(appstore_header)
print('\n')
explore_data(appstore, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


### Exploring Google Play Dataset

Using explore_data() above, we'll print the column names and first three rows of Google Play dataset.

In [4]:
print(gplay_header)
print('\n')
explore_data(gplay, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


## Cleaning

### Column Length Checking

Because there is [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) that one of the data column length is not same as header column length. Let's we check using this loop' how to find the mismatch row.

In [5]:
print ("Apple Store mismatch result :")
for row in appstore:
    if len(row) != len(appstore_header):
        print("Index :", appstore.index(row))
        print(row)
print("\n")
print("Google Play mismatch result :")        
for row in gplay:
    if len(row) != len(gplay_header):
        print("Index :", gplay.index(row))
        print(row)

Apple Store mismatch result :


Google Play mismatch result :
Index : 10472
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


#### Deleting Wrong Data

We didn't found mismatch from App Store data, But in Google Play dataset we found one row that not have same length with it's header. We need delete row[10472] from headerless data.

In [6]:
print("Old length :", len(gplay))

# deleting the mismatch row[10472]
#
del gplay[10472]
print("New length :", len(gplay))

# Re-check 
#
print("\nGoogle Play mismatch result :")        
for row in gplay:
    if len(row) != len(gplay_header):
        print("Index :", gplay.index(row))
        print(row)

Old length : 10841
New length : 10840

Google Play mismatch result :


We have deleted mismatch row and double checking it. Good...! Zero result.

### Duplicate Checking

From the [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion) section, the Google Play have duplicate entries.

In [7]:
duplicate_apps = []
unique_apps = []

for app in gplay:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:\n', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps:
 ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews and remove the other entries for any given app.

#### Removing Duplicate Entries

In [8]:
reviews_max = {}

for app in gplay:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    
    else:
        reviews_max[name] = n_reviews

Inspect the dictionary to make sure everything went as expected. Measure the length of the dictionary — remember that the expected length is 9,659 entries.

In [9]:
print('Expected length: ', len(gplay) - 1181)
print('Actual length: ', len(reviews_max))

Expected length:  9659
Actual length:  9659


Use the dictionary created above to remove the duplicate rows

In [10]:
gplay_nodup = []
already_added = []

for app in gplay:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        gplay_nodup.append(app)
        already_added.append(name)

Explore the android_clean data set to ensure everything went as expected. The data set should have 9,659 rows.

In [11]:
print('Expected length: ', len(gplay) - 1181)
print('Google Play no duplicate length: ', len(gplay_nodup))

Expected length:  9659
Google Play no duplicate length:  9659


### Non-English Checking

We're not interested in keeping these apps, so we'll remove them. One way to go about this is to remove each app with a name containing a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;), and other symbols (+, *, /).

Behind the scenes, each character we use in a string has a corresponding number associated with it. For instance, the corresponding number for character 'a' is 97, character 'A' is 65, and character '爱' is 29,233. We can get the corresponding number of each character using the ord() [built-in function](https://docs.python.org/3/library/functions.html#ord).

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the [ASCII](https://en.wikipedia.org/wiki/ASCII) (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.

In [12]:
# Make ASCII checker function
#
def en_check(string):
    
    for char in string:
        if ord(char) > 127:
            return False
        
    return True

In [13]:
print(en_check('Instagram'))
print(en_check('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(en_check('Docs To Go™ Free Office Suite'))
print(en_check('Instachat 😜'))

True
False
False
False


We saw that the function couldn't correctly identify certain English app names like 'Docs To Go™ Free Office Suite' and 'Instachat 😜'. This is because emojis and characters like ™ fall outside the ASCII range and have corresponding numbers over 127.

In [14]:
print(ord('™'))
print(ord('😜'))

8482
128540


If we're going to use the function we've created, we'll lose useful data since many English apps will be incorrectly labeled as non-English. To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range.

In [15]:
# Make revision for ASCII checker function
#
def en_check(string):
    threshold = 0     
    
    for char in string:
        if ord(char) > 127:
            threshold += 1
    
    if threshold > 3:
        return False
    else:
        return True

In [16]:
print(en_check('Instagram'))
print(en_check('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(en_check('Docs To Go™ Free Office Suite'))
print(en_check('Instachat 😜'))

True
False
True
True


#### Remove Non-English App

Using the new function to filter out non-English apps from both data sets.

In [17]:
appstore_en = []
gplay_en = []

for app in appstore:
    name = app[1]
    if en_check(name):
        appstore_en.append(app)

for app in gplay_nodup:
    name = app[0]
    if en_check(name):
        gplay_en.append(app)

In [18]:
print('English Apple Store Apps :', len(appstore_en))
print('English Goggle Play Apps :', len(gplay_en))

English Apple Store Apps : 6183
English Goggle Play Apps : 9614


### Isolating Free Apps

As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps; we'll need to isolate only the free apps for our analysis.

Isolating the free apps will be our last step in the data cleaning process.

In [19]:
appstore_fin = []
gplay_fin = []

for app in appstore_en:
    price = app[4]
    if price == '0' or price == '0.0':
        appstore_fin.append(app)

for app in gplay_en:
    price = app[7]
    if price == '0' or price == '0.0':
        gplay_fin.append(app)

In [20]:
print('Free Apple Store Apps :', len(appstore_fin))
print('Free Goggle Play Apps :', len(gplay_fin))

Free Apple Store Apps : 3222
Free Goggle Play Apps : 8864


---

# Analyzing

So far, we spent a good amount of time on cleaning data, and:

* Removed inaccurate data
* Removed duplicate app entries
* Removed non-English apps
* Isolated the free apps

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.
Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

## Percentage Table

We'll build two functions we can use to analyze the frequency tables:

* One function to generate frequency tables that show percentages
* Another function we can use to display the percentages in a descending order

In [21]:
def freq_table(dataset, index):
    ''' 
    dataset, expected to be a list of lists
    index, expected to be an integer
    '''
    genres = {}
    total = 0
    percent_table = {}
    
    for row in dataset :
        total += 1
        genre = row[index]
        if genre in genres:
            genres[genre] += 1
        else :
            genres[genre] = 1 
    
    for key in genres:
        percentage = (genres[key] / total) * 100
        percent_table[key] = percentage 
    
    return percent_table

In [22]:
def display_table(dataset, index):
    '''
    Takes in two parameters: dataset and index. dataset is expected to be a list of lists, and index is expected to be an integer.
    Generates a frequency table using the freq_table() function (which you're going to write as an exercise).
    Transforms the frequency table into a list of tuples, then sorts the list in a descending order.
    Prints the entries of the frequency table in descending order.
    '''
    
    table = freq_table(dataset, index)
    table_display = []
    
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Apple Store Percentage Table

We will generate frequency tables for the columns prime_genre from Apple Store 

In [23]:
display_table(appstore_fin, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


The most genre for Apple Store free apps is games with 58.16%, following it entertainment with 7.88%. Both have a number far adrift.  

The general impression is most of the free apps in Apple Store is designed for entertainment instead of practical purposes (such as education, shopping, utilities, productivity, or lifestyle). But the large number of apps for particular genre doesn't imply that apps of that genre generally have a large number of users

### Google Play Percentage Table

#### Categories 

We will generate frequency tables for the columns category from Google Play

In [24]:
display_table(gplay_fin, 1) # Categories

FAMILY : 19.223826714801444
GAME : 9.510379061371841
TOOLS : 8.461191335740072
BUSINESS : 4.580324909747293
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.5424187725631766
SPORTS : 3.4183212996389893
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2490974729241873
HEALTH_AND_FITNESS : 3.068592057761733
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.782490974729242
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.128158844765343
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
ENTERTAINMENT : 0.8799638989169676
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 0

#### Genres

We will generate frequency tables for the columns genres from Google Play

In [25]:
display_table(gplay_fin, 9) # Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.580324909747293
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.5424187725631766
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2490974729241873
Action : 3.1024368231046933
Health & Fitness : 3.068592057761733
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.861462093862816
Video Players & Editors : 1.782490974729242
Casual : 1.7486462093862816
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.925090252707581

The most genre (from 'categories') is family with 19.22%.  
And overal genre in Google Play percentage table is more evenly distributed compared to Apple Store percentage table.

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

## Most Popular Apps by Genre

The frequency tables we analyzed on the previous screen showed us that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps. Now, we'd like to get an idea about the kind of apps with the most users.  

### Most Popular Apps in Apple Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

In [26]:
genres_appstore = freq_table(appstore_fin, -5)

for genre in genres_appstore:
    total = 0
    len_genre = 0
    for app in appstore_fin:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


#### Analysis

In [27]:
# 1st place, avg_rating : 86090.33333333333
#
for app in appstore_fin:
    if app[-5] == 'Navigation':
        print(app[1], ' : ', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic  :  345046
Google Maps - Navigation & Transit  :  154911
Geocaching®  :  12811
CoPilot GPS – Car Navigation & Offline Maps  :  3582
ImmobilienScout24: Real Estate Search in Germany  :  187
Railway Route Search  :  5


The Navigation category was dominated by Waze and Google, which has half a million users together. Not a ideal category for new apps.

In [28]:
# 2nd place, avg_rating : 74942.11111111111
#
for app in appstore_fin:
    if app[-5] == 'Reference':
        print(app[1], ' : ', app[5])

Bible  :  985920
Dictionary.com Dictionary & Thesaurus  :  200047
Dictionary.com Dictionary & Thesaurus for iPad  :  54175
Google Translate  :  26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran  :  18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition  :  17588
Merriam-Webster Dictionary  :  16849
Night Sky  :  12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE)  :  8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools  :  4693
GUNS MODS for Minecraft PC Edition - Mods Tools  :  1497
Guides for Pokémon GO - Pokemon GO News and Cheats  :  826
WWDC  :  762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free  :  718
VPN Express  :  14
Real Bike Traffic Rider Virtual Reality Glasses  :  8
教えて!goo  :  0
Jishokun-Japanese English Dictionary & Translator  :  0


In [29]:
# 3rd place, avg_rating : 74942.11111111111
#
for app in appstore_fin:
    if app[-5] == 'Business':
        print(app[1], ' : ', app[5])

Indeed Job Search  :  38681
Flashlight ◎  :  24744
Adobe Acrobat Reader: View, Create, & Convert PDFs  :  20069
Scanner App - PDF Document Scan  :  11696
SayHi Translate  :  8623
ADP Mobile Solutions  :  8324
Sideline - 2nd Phone Number  :  7907
Uber Driver  :  3289
AirWatch Agent  :  1150
VPN Go - Safe Fast & Stable VPN Proxy  :  881
Cisco AnyConnect  :  825
GreenVPN - Free & fast VPN with unlimited traffic  :  464
iPlum Business Phone Number for Calling & Texting  :  392
OPEN Forum  :  200
Pulse Secure  :  53
DingTalk  :  40
Mon Espace - Pôle emploi  :  11


In [30]:
# 4th place, avg_rating : 71548.34905660378
#
for app in appstore_fin:
    if app[-5] == 'Social Networking':
        print(app[1], ' : ', app[5])

Facebook  :  2974676
Pinterest  :  1061624
Skype for iPhone  :  373519
Messenger  :  351466
Tumblr  :  334293
WhatsApp Messenger  :  287589
Kik  :  260965
ooVoo – Free Video Call, Text and Voice  :  177501
TextNow - Unlimited Text + Calls  :  164963
Viber Messenger – Text & Call  :  164249
Followers - Social Analytics For Instagram  :  112778
MeetMe - Chat and Meet New People  :  97072
We Heart It - Fashion, wallpapers, quotes, tattoos  :  90414
InsTrack for Instagram - Analytics Plus More  :  85535
Tango - Free Video Call, Voice and Chat  :  75412
LinkedIn  :  71856
Match™ - #1 Dating App.  :  60659
Skype for iPad  :  60163
POF - Best Dating App for Conversations  :  52642
Timehop  :  49510
Find My Family, Friends & iPhone - Life360 Locator  :  43877
Whisper - Share, Express, Meet  :  39819
Hangouts  :  36404
LINE PLAY - Your Avatar World  :  34677
WeChat  :  34584
Badoo - Meet New People, Chat, Socialize.  :  34428
Followers + for Instagram - Follower Analytics  :  28633
GroupMe  :  

Same as Navigation, Social Networking category is dominated by few big company, such as Facebook, Pinterest, Skype, and Messenger. Not an ideal option for new apps.

In [31]:
# 5th place, avg_rating : 7003.983050847458
#
for app in appstore_fin:
    if app[-5] == 'Education':
        print(app[1], ' : ', app[5])

Duolingo - Learn Spanish, French and more  :  162701
Guess My Age  Math Magic  :  123190
Lumosity - Brain Training  :  96534
Elevate - Brain Training and Games  :  58092
Fit Brains Trainer  :  46363
ClassDojo  :  35440
Memrise: learn languages  :  20383
Peak - Brain Training  :  20322
Canvas by Instructure  :  19981
ABCmouse.com - Early Learning Academy  :  18749
Quizlet: Study Flashcards, Languages & Vocabulary  :  16683
Photomath - Camera Calculator  :  16523
iTunes U  :  15801
Blackboard Mobile Learn™  :  13567
Star Chart  :  13482
Remind: Fast, Efficient School Messaging  :  9796
PBS KIDS Video  :  8651
Toca Kitchen Monsters  :  8062
Toca Hair Salon - Christmas Gift  :  8049
Edmodo  :  7197
Prodigy Math Game  :  6683
Epic! - Unlimited Books for Kids  :  6676
ChineseSkill -Learn Mandarin Chinese Language Free  :  6077
Google Classroom  :  5942
TED  :  5782
Khan Academy: you can learn anything  :  5459
Got It - Homework Help Math, Chem, Physics Solver  :  4903
PowerSchool Mobile  : 

### Recommendation

From the top 5 installation genres, we find that navigation and social networking are dominated by several large companies. I think it's a bit heavy for new applications to compete.  

As an alternative, we can create new applications with the Reference, Business, or Education categories, because in these categories, the makers tend to vary. Moreover, many ideas can be developed from these three categories. Such as making books, business planning or fresh educational platforms.

### Most Popular Apps in Apple Store

In the previous screen, we came up with an app profile recommendation for the App Store based on the number of user ratings. We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [32]:
display_table(gplay_fin, 5)

1,000,000+ : 15.749097472924186
100,000+ : 11.563628158844766
10,000,000+ : 10.503158844765343
10,000+ : 10.209837545126353
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, we'll need to convert each install number from string to float. This means we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error.

In [35]:
categories_android = freq_table(gplay_fin, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in gplay_fin:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1704192.3399014778
COMICS : 817657.2727272727
COMMUNICATION : 38326063.197916664
DATING : 854028.8303030303
EDUCATION : 1768500.0
ENTERTAINMENT : 9146923.076923076
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4167457.3602941176
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 12914435.883748516
FAMILY : 5180161.789906103
MEDICAL : 123064.7898089172
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 4274688.722772277
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16772838.591304347
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24790074.17721519
NEWS_AND_MAGAZINES : 

The most installed category is COMMUNICATION  by avegrage score 38,326,063.197916664. This number is heavily skewed by giants, such as Messenger, Gmail, WhatsApp and Skype.

In [36]:
for app in gplay_fin:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Messenger – Text and Video Chat for Free : 1,000,000,000+
Gmail : 1,000,000,000+
imo beta free calls and text : 100,000,000+
imo free video calls and chat : 500,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
WhatsApp Messenger : 1,000,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Hangouts : 1,000,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

The runner up is VIDEO_PLAYERS category by 24,790,074.17721519 as average. This number is dominated by Google Play Movies & TV and Youtube.

In [37]:
for app in gplay_fin:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Motorola Gallery : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
YouTube : 1,000,000,000+
Dubsmash : 100,000,000+
MX Player : 500,000,000+
VLC for Android : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


Then we're going to analyze BOOKS_AND_REFERENCE category which is have 87,67,811.894736841 as average installation. 

In [38]:
for app in gplay_fin:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [39]:
for app in gplay_fin:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [40]:
for app in gplay_fin:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

### Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.